### PR TITLE
Add basic authentication usage example.

### PR DESCRIPTION
**Pull request by normalex**
This shall be a beneficial example for others as well.

### CODE REVIEW

**Comment by prat0318** [comment]
Thanks for the PR. I think we can shorten down the example by following the same structure as the previous pet example. Also, can we remove vendor specific names like circonus.

**Comment by normalex** [comment]
Done.

File: README.rst [hunk 1]
```diff
@@ -43,6 +43,30 @@ Example Usage
...
     pet = client.pet.getPetById(petId=42).result()
 
+Example with Basic Authentication
+-------------------------
+
+.. code-block:: python
+
+    from bravado.requests_client import RequestsClient
+    from bravado.client import SwaggerClient
+    from bravado.swagger_model import load_file
```
Comment (laucia) [inline README.rst]: import is unused

File: README.rst [hunk 2]
```diff
@@ -43,6 +43,30 @@ Example Usage
...
+    
+    http_client = RequestsClient()
+    http_client.set_basic_auth(
+        'api.yourhost.com',
+        'username', 'password'
+    )
+    # some APIs, require explicitly set header "Accept: application/json"
```
Comment (laucia) [inline README.rst]: I am not sure I understand how this is directly related to authentication, should this really be in the default standard example?

**Comment by sjaensch** [comment]
I merged this after addressing @laucia's comments and doing minor stylistic changes.

**Pull request merged**